docsite/rst/developing_plugins.rst

```diff
@@ -9,10 +9,26 @@ in ansible and were most convenient to be made pluggable.
 This section will explore these features, though they are generally not common in terms of things people would look to extend quite
 as often.
 
-.. _developing_connection_type_plugins:
+.. _developing_modules:
 
-Connection Type Plugins
------------------------
+Task Plugins (aka Modules)
+--------------------------
+
+This is the most common type of plugin people need and develop, so we have a dedicated page for this type. See :doc:`developing_modules`.
+
+.. _developing_action_plugins:
+
+Action Plugins
+--------------
+
+Action plugins are actually a 'front' to modules, if they match the module name, they get executed instead and are responsible for using the module itself, if needed. These are used for when part of the task (or all of it) needs to happen on the machine running Ansible. raw, script, copy and template are examples of action plugins, in the case of template, the module itself is just a documentation holder, all the work is done by the action plugin.
```

Comment:
Action plugins are actually a 'front' to modules.  If an action plugin matches a module name, then the action plugin is executed instead and it will use the module by the same name, if needed. Action plugins are used for when part of the task (or all of it) needs to happen on the machine running Ansible. Examples of action plugins include 'raw', 'script', 'copy', and 'template' are examples of action plugins. In the case of 'template', the module itself is just a documentation holder and all of the work is done by the action plugin.